app/storage/storageFactory.js

```diff
@@ -64,8 +64,17 @@ class StorageFactory {
     this.logger = (logger || initLogger);
   }
 
-  newResourceHandler(resourceKey, bucketName, location) {
-    const targetHandler = config.defaultHandler.constructor(this.logger, resourceKey, bucketName, location);
+  newResourceHandler(args={}) {
+    var { path, bucketConfObj, location, endpoint, org } = args;
+    // if(!path || !bucketConfObj){
+    //   throw new Error(`required args: { path, bucketConfObj }`);
+    // }
+    console.log(44444, { path, bucketConfObj, location, endpoint, org })
+    // args is an obj of other attrs to pass that can be useful in certain contexts, such as { org }
+    const targetHandler = config.defaultHandler.constructor({
+      ..._.pick(this, ['logger']),
+      ...args
```

Comment:
This is interesting syntax, but (to me at least) not intuitive.  A code comment explaining what it's doing, for newbs like me, would be appreciated.